operator/pkg/reconciliation/handler.go

```diff
@@ -153,7 +153,17 @@ func (rc *ReconciliationContext) calculateReconciliationActions() (reconcile.Res
 		return result.Error(err).Output()
 	}
 
-	return rc.ReconcileAllRacks()
+	result, err := rc.ReconcileAllRacks()
+
+	if err == nil {
+		// Update PSP status
+		// Always sync the datacenter status with the PSP health status
+		if err := rc.PSPHealthUpdater.Update(*rc.Datacenter); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
Any time we may have updated the status, we want to sync that change with the PSP health status. However, we also want to preserve the result from `ReconcileAllRacks()` so that requeue results are not lost.